### PR TITLE
RD-4792 RD-4793 Add snapshot support for page groups and widgets

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added snapshot support for page groups and widgets

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/4216/

 

## Documentation
N/A

[RD-1234]: https://cloudifysource.atlassian.net/browse/RD-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by qooban** [comment]
I'll review it tomorrow.

File: backend/routes/Snapshots.ts [hunk 1]
```diff
@@ -51,13 +53,14 @@ router.use(express.json());
...
-(function templates() {
-    const propertiesToOmit = ['name', 'custom'] as const;
-    type PropertiesToOmit = typeof propertiesToOmit[number];
-    type TemplatesSnapshot = Omit<Template, PropertiesToOmit>[];
+const commonPropertiesToOmit = ['custom'] as const;
+type CommonPropertiesToOmit = typeof commonPropertiesToOmit[number];
 
+const templatePropertiesToOmit = [...commonPropertiesToOmit, 'name'] as const;
+export type TemplatesSnapshot = Omit<Template, typeof templatePropertiesToOmit[number]>[];
+(() => {
```
Comment (qooban) [inline backend/routes/Snapshots.ts:61]: General:

Why we use IIFE to define routes in this file?

**Review by qooban — APPROVED** [review]
Looks good 👍🏻 

// I'm not capable (high workload) of doing very detailed review //

File: backend/routes/Snapshots.ts (hunk 1 above)
Comment (kubama) [reply backend/routes/Snapshots.ts:61]: My intention was to group endpoints into related get/restore pairs for higher clarity and code navigability.

Comment (qooban) [reply backend/routes/Snapshots.ts:61]: Got it.

**Pull request merged**